src/main/java/calibration/CalibrationApp.java

```diff
@@ -194,18 +194,21 @@ public void setPersistedFile(Optional<File> file) {
      * Loads person data from the specified file. The current data will be replaced.
      *
      * @param file
+     * @param headlessCompute true if computing headless, false if computing from the UI
      */
-    public void loadAnalysisFromFile(File file) {
+    public void loadAnalysisFromFile(File file, boolean headlessCompute) {
         try {
             JAXBContext context = JAXBContext.newInstance(Analysis.class);
             Unmarshaller um = context.createUnmarshaller();
 
             // Reading XML from the file and unmarshalling.
             this.analysis = (Analysis) um.unmarshal(file);
 
-            // Save the file path to the registry.
-            setAnalysisFile(Optional.ofNullable(file));
-            setPersistedFile(Optional.ofNullable(file));
+            if (!headlessCompute) {
+                // Save the file path to the registry.
+                setAnalysisFile(Optional.ofNullable(file));
```

Comment:
That code that writes to the registry is pretty antiquated. I would do something similar to what we have in HMS `FileBrowseButton::setPersistedBrowseLocation` and `getPersistedBrowseLocation`